pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -503,25 +504,15 @@ func (qjm *XController) GetQueueJobsEligibleForPreemption() []*arbv1.AppWrapper
 
 			if (int(value.Status.Running) + int(value.Status.Succeeded)) < replicas {
 
-				//Check to see if if this AW job has been dispatched for a time window before preempting
-				conditionsLen := len(value.Status.Conditions)
-				var dispatchConditionExists bool
-				dispatchConditionExists = false
-				var condition arbv1.AppWrapperCondition
-				// Get the last time the AppWrapper was dispatched
-				for i := (conditionsLen - 1); i > 0; i-- {
-					condition = value.Status.Conditions[i]
-					if condition.Type != arbv1.AppWrapperCondDispatched {
-						continue
-					}
-					dispatchConditionExists = true
-					break
-				}
+				// Check for the minimum age and then skip preempt if current time is not beyond minimum age
+				// The minimum age is controlled by the requeuingTimeMinutes stanza
+				// For preemption, the time is compared to the last condition in the AppWrapper
+				condition := value.Status.Conditions[len(value.Status.Conditions) - 1]
```

Comment:
should we check if last condition is running?

Reply:
Done. Now it checks for the latest condition time wise. Once the AppWrapper gets dispatched again, it is now using that time since that is the most recent time.